src/augeas.c

```diff
@@ -2020,6 +2021,35 @@ int aug_escape_name(augeas *aug, const char *in, char **out) {
     return result;
 }
 
+int aug_load_file(struct augeas *aug, const char *file) {
+    int result = -1;
+    struct tree *meta = tree_child_cr(aug->origin, s_augeas);
+    struct tree *load = tree_child_cr(meta, s_load);
+    bool found = false;
+
+    api_entry(aug);
+
+    ERR_NOMEM(load == NULL, aug);
+
+    list_for_each(xfm, load->children)  {
+      if (filter_matches(xfm, file)) {
+        transform_load(aug, xfm, file);
```

Comment:
One thing I noticed in my testing is that the tree thet gets created is marked dirty; issuing `save` right after loading a file will try to save the file, which is not correct. I'll provide a patch that shows how to call `tree_clean` to mark the subtree as clean and therefore avoid that.